src/material/radio/radio.ts

```diff
@@ -807,4 +808,13 @@ export class MatRadioButton implements OnInit, AfterViewInit, DoCheck, OnDestroy
       }
     }
   }
+
+  protected getInputRequiredAttribute(): boolean | null {
+    // we never want to set required attribute on input when we have MatRadioGroup as we will set
+    // aria-required directly on MatRadioGroup if its required as setting on all MatRadioButton for
+    // it's MatRadioGroup would be confusing for assistive technology.
+    if (this.radioGroup) return null;
```

Comment:
Why is this `null` instead of `false`, but when we're setting it according to `this.required` we don't do `|| null`? Is there a difference between `aria-required="false"` and just not having the attribute?